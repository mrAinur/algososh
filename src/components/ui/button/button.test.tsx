import renderer from "react-test-renderer";
import { Button } from "./button";
import { render, screen, fireEvent } from "@testing-library/react";

describe("Тест компонента button", () => {
  it("с текстом", () => {
    const button = renderer
      .create(<Button text="Тест текста в кнопке" />)
      .toJSON();
    expect(button).toMatchSnapshot();
  });
  it("без текста", () => {
    const button = renderer.create(<Button />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("- заблокированной кнопки", () => {
    const button = renderer.create(<Button disabled={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("с индикацией загрузки", () => {
    const button = renderer.create(<Button isLoader={true} />).toJSON();
    expect(button).toMatchSnapshot();
  });
  it("на корректность вызова функции", () => {
    const getCallFunction = jest.fn();
    render(<Button onClick={getCallFunction} />);
    const button = screen.getByRole("button");
    fireEvent.click(button);
    expect(getCallFunction).toBeCalledTimes(1);
  });
});
