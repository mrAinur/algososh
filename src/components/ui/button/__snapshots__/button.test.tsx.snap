// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тест компонента button - заблокированной кнопки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест компонента button без текста 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тест компонента button с индикацией загрузки 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тест компонента button с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Тест текста в кнопке
  </p>
</button>
`;
